catalog/src/main/java/com/qubole/quark/catalog/db/SchemaFactory.java

```diff
@@ -109,23 +109,30 @@ public QuarkFactoryResult create(Properties info) throws QuarkException {
       List<JdbcSource> jdbcSources = jdbcSourceDAO.findByDSSetId(dsSetId);
```

Comment:
I couldn't understand this part... What would happen here if default data source is set null in dsset table?